### PR TITLE
Move support to OTP 22+

### LINKED ISSUE #103: Drop OTP =<21 support
Ref: #93 #51 #54 ...

### PR DESCRIPTION
**Pull request by paulo-ferraz-oliveira**
Closes #103.

This pull request is being split into several other pull requests, to ease review. These are listed below.

* https://github.com/elli-lib/elli/pull/119
* https://github.com/elli-lib/elli/pull/120
* https://github.com/elli-lib/elli/pull/122

### CODE REVIEW

File: elvis.config
Comment (paulo-ferraz-oliveira) [inline elvis.config:1]: Work in blacklist mode (accept all defaults except...), and add `test` to the mix.

File: src/elli_http.erl [hunk 1]
```diff
@@ -58,7 +60,7 @@ start_link(Server, ListenSocket, Options, Callback) ->
       Options      :: proplists:proplist(),
       Callback     :: elli_handler:callback().
 accept(Server, ListenSocket, Options, Callback) ->
-    case catch elli_tcp:accept(ListenSocket, Server, accept_timeout(Options)) of
+    case elli_tcp:accept(ListenSocket, Server, accept_timeout(Options)) of
```
Comment (paulo-ferraz-oliveira) [inline src/elli_http.erl:63]: I don't see where `catch` was actually used, since all the clauses derive not from exceptions, but simple function returns.

**Comment by paulo-ferraz-oliveira** [comment]
~CI failing for OTP 26, 24, and 22. I'll try to replicate locally.~

**Comment by paulo-ferraz-oliveira** [comment]
~It's possible the failure for non-OTP 26 is due to #99, since re-executing the tests (in a fork) showed ✅.~

**Comment by paulo-ferraz-oliveira** [comment]
~Removed OTP 26 from scope, since there're SSL -related issues to solve. Should be moved to a new PR, I guess...~

**Comment by tsloughter** [comment]
Woo, these are all looking good.

And yea, for 26 the ssl certs may need to be re-created. I had to do this for chatterbox tests too.

**Comment by paulo-ferraz-oliveira** [comment]
Is anything lacking for this one to get merged? I can then rebase and update the other ones.

I think I'll deal with OTP 26 in a separate PR so as to not inflict further confusion in these pull requests... 😄

**Comment by paulo-ferraz-oliveira** [comment]
@tsloughter, if you prefer I can also reduce the scope for this and do multiple pull requests to an isolated branch until you're fully satisfied and then we merge to the main branch.

**Comment by tsloughter** [comment]
This doesn't look too bad, I just hadn't taken a close look.

**Comment by paulo-ferraz-oliveira** [comment]
@tsloughter, what's required to push this forward?

**Comment by paulo-ferraz-oliveira** [comment]
👋

**Comment by paulo-ferraz-oliveira** [comment]
@tsloughter, what's required to push this forward?

**Comment by paulo-ferraz-oliveira** [comment]
@tsloughter, what's required to push this forward?

**Comment by tsloughter** [comment]
Hey sorry, it is just a large PR. But it looks like a lot of it is formatting?

**Comment by paulo-ferraz-oliveira** [comment]
👋 it's probably the commit related to Elvis, but I can try to break it down further.

**Comment by paulo-ferraz-oliveira** [comment]
I'm moving to draft, to be able to break it down into several pull requests. Once that's done, I'll close this one without it being merged.

**Comment by paulo-ferraz-oliveira** [comment]
All new pull requests mentioned in the top description.

**Pull request closed without merging**